src/Cli/func/Actions/HostActions/StartHostAction.cs

```diff
@@ -300,7 +300,7 @@ private async Task<IDictionary<string, string>> GetConfigurationSettings(string
 
             // when running locally in CLI we want the host to run in debug mode
             // which optimizes host responsiveness
-            settings.Add("AZURE_FUNCTIONS_ENVIRONMENT", "Development");
+            settings.TryAdd("AZURE_FUNCTIONS_ENVIRONMENT", "Development");
```

Comment:
There are a lot of side effects if we allow for this to be overridden. The WebJobs SDK and host behaviors for polling intervals, logging, and other things will change, and this could significantly impact the local debugging experience in surprising ways.